src/charm.py

```diff
@@ -363,12 +369,25 @@ def _on_leader_elected(self, event: LeaderElectedEvent) -> None:
         except RetryError:
             pass  # This error can happen in the first leader election, as Patroni is not running yet.
 
+    def _create_pgdata(self, container: Container):
+        """Create the PostgreSQL data directory."""
+        path = f"{self._storage_path}/pgdata"
+        if not container.exists(path):
+            container.make_dir(
+                path, permissions=0o700, user=WORKLOAD_OS_USER, group=WORKLOAD_OS_GROUP
```

Comment:
I think we should set 770. It will reduce the number of issues on OpenShift
```suggestion
                path, permissions=0o770, user=WORKLOAD_OS_USER, group=WORKLOAD_OS_GROUP
```